libsql/src/replication/mod.rs

```diff
@@ -95,27 +98,76 @@ impl Writer {
 #[derive(Clone)]
 pub(crate) struct EmbeddedReplicator {
     replicator: Arc<Mutex<Replicator<Either<RemoteClient, LocalClient>>>>,
+    bg_abort: Option<Arc<DropAbort>>,
 }
 
 impl EmbeddedReplicator {
-    pub async fn with_remote(client: RemoteClient, db_path: PathBuf, auto_checkpoint: u32, encryption_key: Option<bytes::Bytes>) -> Self {
+    pub async fn with_remote(
+        client: RemoteClient,
+        db_path: PathBuf,
+        auto_checkpoint: u32,
+        encryption_key: Option<bytes::Bytes>,
+        perodic_sync: Option<Duration>,
+    ) -> Self {
         let replicator = Arc::new(Mutex::new(
-            Replicator::new(Either::Left(client), db_path, auto_checkpoint, encryption_key)
-                .await
-                .unwrap(),
+            Replicator::new(
+                Either::Left(client),
+                db_path,
+                auto_checkpoint,
+                encryption_key,
+            )
+            .await
+            .unwrap(),
         ));
 
-        Self { replicator }
+        let mut replicator = Self {
+            replicator,
+            bg_abort: None,
+        };
+
+        if let Some(sync_duration) = perodic_sync {
+            let replicator2 = replicator.clone();
+
+            let jh = tokio::spawn(
+                async move {
+                    loop {
+                        if let Err(e) = replicator2.sync_oneshot().await {
+                            tracing::error!("replicator sync error: {}", e);
+                        }
+
+                        tokio::time::sleep(sync_duration).await;
+                    }
+                }
+                .instrument(tracing::info_span!("periodic_sync")),
+            );
+
+            replicator.bg_abort = Some(Arc::new(DropAbort(jh.abort_handle())));
+        }
```

Comment:
would it make sense to spawn this task in a join set instead?

Reply:
That is an option, I think the behavior is the same here just saves us some code. So maybe we can refactor to that in the future.